crates/std_detect/src/detect/os/linux/aarch64.rs

```diff
@@ -6,6 +6,23 @@ use crate::detect::{bit, cache, Feature};
 /// Try to read the features from the auxiliary vector, and if that fails, try
 /// to read them from /proc/cpuinfo.
 pub(crate) fn detect_features() -> cache::Initializer {
+    #[cfg(all(target_arch = "aarch64", target_os = "android"))]
+    {
+        // Samsung Exynos 9810 has a bug that big and little cores have different
+        // ISAs. And on older Android (pre-9), the kernel incorrectly reports
+        // that features available only on some cores are available on all cores.
+        // https://reviews.llvm.org/D114523
+        let mut arch = [0_u8; libc::PROP_VALUE_MAX as usize];
+        let len = unsafe {
+            libc::__system_property_get(
+                b"ro.arch\0".as_ptr() as *const libc::c_char,
+                arch.as_mut_ptr() as *mut libc::c_char,
+            )
+        };
+        if len > 0 && arch.starts_with(b"exynos9810\0") {
```

Comment:
Is this checking that the returned string starts with `exynos9810` or that it is exactly equal to `exynos9810`? In the former case you wouldn't want the `\0` in the string.

Reply:
I originally meant the latter, but looking at the LLVM patch they seem to be using the former.

So, I removed `\0` to match the LLVM patch's behavior.